src/core_landice/mode_forward/mpas_li_calving.F

```diff
@@ -1366,16 +1384,28 @@ subroutine von_Mises_calving(domain, err)
              if ( .not. li_mask_is_floating_ice(cellMask(iCell)) ) then
                 calvingVelocity(iCell) = sqrt(xvelmean(iCell)**2.0_RKIND + yvelmean(iCell)**2.0_RKIND) * &
                                          vonMisesStress(iCell) / config_grounded_von_Mises_threshold_stress
-             elseif ( li_mask_is_floating_ice(cellMask(iCell)) ) then
-                calvingVelocity(iCell) = sqrt(xvelmean(iCell)**2.0_RKIND + yvelmean(iCell)**2.0_RKIND) * &
+             elseif ( li_mask_is_floating_ice(cellMask(iCell)) .and. config_floating_von_Mises_threshold_stress .ne. 0.0_RKIND) then
+                calvingVelocity(iCell) = sqrt(xvelmean(iCell)**2 + yvelmean(iCell)**2) * &
                                          vonMisesStress(iCell) / config_floating_von_Mises_threshold_stress
              endif
           enddo
 
           ! Convert calvingVelocity to calvingThickness
-          call apply_calving_velocity(meshPool, geometryPool, velocityPool, domain, err)
-
+          call apply_front_ablation_velocity(meshPool, geometryPool, velocityPool, applyToGrounded, &
+                                      applyToFloating, applyToGroundingLine, applyCalving, applyMelting, domain, err)
+
+
+          ! If floating VM threshold is zero, remove floating ice.
+          if ( config_floating_von_Mises_threshold_stress .eq. 0.0_RKIND ) then
+             do iCell = 1,nCells
+                if ( li_mask_is_floating_ice(cellMask(iCell)) .and. &
```

Comment:
Thinking about this more carefully, we may need to be more careful at high res.  We sometimes get 'subglacial lakes' a bit behind the GL where ice goes afloat in isolated pockets.  We wouldn't want to zap those spots.  The existing float kill routine also suffers from this issue.  We could add a TODO item for this for now and deal with it in a later PR if/when we run into the issue operationally.

I think the solution would be to do a flood-fill algorithm from all open ocean cells through all floating cells.  Because that is a lot more complicated than the code here, we might want to write that in a way that we implement that in the float-kill routine and call the float-kill routine from here.

Reply:
Please add the TODO item mentioned above.  Also, looking at this again, I think this requires more careful treatment of the dynamic ice mask.  We don't want to leave a stranded band of nondynamic floating ice at the former calving front, but I guess we also don't want to delete nondynamic floating ice attached to grounded ice.